report_xlsx/models/ir_report.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2015 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ReportAction(models.Model):
+    _inherit = "ir.actions.report"
+
+    report_type = fields.Selection(
+        selection_add=[("xlsx", "XLSX")], ondelete={"xlsx": "set default"}
+    )
+
+    @api.model
+    def _render_xlsx(self, report_ref, docids, data):
+        report_sudo = self._get_report(report_ref)
+        report_model_name = "report.%s" % report_sudo.report_name
+        report_model = self.env[report_model_name]
+        return (
+            report_model.with_context(active_model=report_sudo.model)
+            .sudo(False)
+            .create_xlsx_report(docids, data)  # noqa
+        )
+
+    @api.model
+    def _get_report_from_name(self, report_name):
+        res = super(ReportAction, self)._get_report_from_name(report_name)
```

Comment:
Apply PEP 3135 (28-Apr-2007) to "super()."

Reply:
Done